File: spec/python/test_eof_exception_bytes.py
import unittest
from eof_exception_bytes import EofExceptionBytes
import kaitaistruct

class TestEofExceptionBytes(unittest.TestCase):
    def test_eof_exception_bytes(self):
        with self.assertRaisesRegexp(EOFError, "^requested \d+ bytes, but only \d+ bytes available$"):
            with EofExceptionBytes.from_file('src/term_strz.bin') as r:
                pass

File: spec/lua/test_eof_exception_bytes.lua
local luaunit = require("luaunit")

require("eof_exception_bytes")

TestEofExceptionBytes = {}

function TestEofExceptionBytes:test_eof_exception_bytes()
    luaunit.assertErrorMsgMatches(".+: requested %d+ bytes, but got only %d+ bytes", EofExceptionBytes.from_file, EofExceptionBytes, "src/term_strz.bin")
end
